python/gcp_client/hydrotools/gcp_client/gcp.py

```diff
@@ -126,13 +61,23 @@ class NWMDataService:
     on Google Cloud Platform.
     """
 
-    def __init__(self, bucket_name='national-water-model', max_processes=None):
+    def __init__(
+        self, 
+        bucket_name: str = 'national-water-model', 
+        max_processes: int = None,
+        default_site_map: pd.DataFrame = None
```

Comment:
Point taken. I ended up calling it `map` because that's the method I used to translate from one location identifier to another. However, this parameter can be used to map (verb) all sorts of extra metadata. Pandas calls it a "mapping correspondence" and the parameter name is the ever so helpful `arg`.  Internally, we often refer to it as a crosswalk, but only in the context of location identifiers. In the past, I've simply referred to it as `metadata` or `external_metadata`.

For now, I've changed it to `location_metadata_mapping`. This makes sure to include the word "metadata" and discards the confusing use of "map" as a verb, since the parameter is a thing (noun) and not an action (verb). It also makes explicit that these metadata are site or location based.

Reply:
Thanks for the background on your thinking and the mention of "mapping correspondence". Ive not run across that language before, so ill add it to my lexicon!

> This makes sure to include the word "metadata" and discards the confusing use of "map" as a verb, since the parameter is a thing (noun) and not an action (verb). It also makes explicit that these metadata are site or location based.

This is not directly related, but may be a good candidate for the git/python packaging/unittesting/"best practices" talk in the future. I am thinking a brief section: argument naming conventions or something along those lines that says exactly what you did above regarding nouns and verbs.